### ISSUE TITLE
German letter transcriptions

### ISSUE THREAD

**Issue by DocOellerson** [lexica/lexica#227]
“SS“ is not a single letter in German. The correct transcription for the letter eszett (“scharfes s“) is “ss“, which should count as to single letters. The capital eszett, which was introduced in 2017, is still very uncommon in daily use and would hinder gameplay, IMHO. See https://en.m.wikipedia.org/wiki/%C3%9F for details.

The letter "Œ" is not used in German. The only word where it is used sometimes is "Œuvre", which commonly is transcribed as "Oeuvre" and is a French loanword.

**Comment by d-mal** [comment]
Yes, this has been talked about a few times already, see #71, #126, and at the end of #69

**Comment by pserwylo** [comment]
Hi all. Can I please get some feedback from any native German speakers? I have a version ready to release which includes two German dictionaries:

    German (Extended) - Includes diacritics: aÅäÄbcdefghijklmnoöÖŒpqrsŠßtuüÜvwxyz
    German

The extended version uses the same dictionary as the current German dictionary in Lexica (but will be renamed to "German (Extended)". It will also include the description "Includes diacritics: aÅäÄbcdefghijklmnoöÖŒpqrsŠßtuüÜvwxyz" alongside the language name when choosing which language to play with.

The new dictionary will just be called "German" and will not have any description next to it.

**I am after some feedback about whether I am doing the right thing with the new  "German" dictionary:**

* An analysis of all unique characters in the current German dictionary shows these: `aÅäÄbcdefghijklmnoöÖŒpqrsŠßtuüÜvwxyz`
* Based on feedback from @d-mal in [this comment](https://github.com/lexica/lexica/issues/71#issuecomment-685184402), I am eplacing ä/Ä->ae, ö/Ö->oe, ü/Ä->ue, ß->ss. Note they are all ending up as lowercase because that is how Lexica stores dictionaries. Upon rendering them to screen, they will all be shown as upper case.
* Based on feedback from @DocOellerson in this issue, I'll replace Œ as oe. Indeed as mentioned above, this is only used in "Oeuvre" and "Oeuvres" in the current dictionary
* I will then remove any duplicate words in case this process accidentally introduced them.
* This leaves the following unique characters in the new dictionary: `aÅbcdefghijklmnopqrsŠtuvwxyz`.

**My questions are:**
* **Does the above sound reasonable?**
* **What should be done with `Å` and `Š` (if anything?)**
* **Is it strictly incorrect to show them as upper case on the gameboard? Or is that just a matter of opinion?** Lexica does have the option of customizing the display of letters on a per-dictionary basis if we need to make these lower case but upper case other languages, but I'd rather leave it consistent if it is a correct option to do so.
* **Is there anything else we should do to improve this new slimline dictionary?**
 

For those technically minded, this is the script used to remove such characters is shown below (could probably be done better, but may as well just hack something together for our purposes) and it will be checked into this repository when done:

```
echo "Copying source dictionary containing diacritics from $DICT_FILE_DIACRITICS..."
cp "$DICT_FILE_DIACRITICS" "$DICT_FILE"
cp "$LICENSE_FILE_DIACRITICS" "$LICENSE_FILE"

echo "Replacing ä/Ä->ae, ö/Ö->oe, ü/Ä->ue, ß->ss, Œ->oe..."
sed -i "s/ä/ae/g" "$DICT_FILE"
sed -i "s/Ä/ae/g" "$DICT_FILE"
sed -i "s/ö/oe/g" "$DICT_FILE"
sed -i "s/Ö/oe/g" "$DICT_FILE"
sed -i "s/ü/ue/g" "$DICT_FILE"
sed -i "s/Ü/ue/g" "$DICT_FILE"
sed -i "s/ß/ss/g" "$DICT_FILE"
sed -i "s/Œ/oe/g" "$DICT_FILE"
```

**Comment by DocOellerson** [comment]
Hi Peter! 

The only valid diacritics in German are öäüß. 
No German word contains Å, Š or Œ, the latter we already talked about.
Words containing these characters are foreign words and are usually not valid in word games. You should remove them from the dictionary.

Some German word games use the umlauts öäü (e.g. Scrabble), crosswords usually use the transcriptions oe, ae, ue. It's a matter of taste what you should use. A boolean switch to choose the method in settings would be an awesome feature. 
Transcribing ß to ss should be considered standard in any German word game.

Concerning upper/lower case: any word can be spelled upper case in German, there is no need for additional lower case letters.

So my suggested character set for German is: 
ABCDEFGHIJKLMNOPQRSTUVWXYZÄÖÜ


Hope this helps! 

Elmar



Am 21. März 2021 21:25:21 MEZ schrieb Peter Serwylo ***@***.***>:
>Hi all. Can I please get some feedback from any native German speakers?
>I have a version ready to release which includes two German
>dictionaries:>
>>
>German (Extended) - Includes diacritics:
>aÅäÄbcdefghijklmnoöÖŒpqrsŠßtuüÜvwxyz>
>    German>
>>
>The extended version uses the same dictionary as the current German
>dictionary in Lexica (but will be renamed to "German (Extended)". It
>will also include the description "Includes diacritics:
>aÅäÄbcdefghijklmnoöÖŒpqrsŠßtuüÜvwxyz" alongside the language name when
>choosing which language to play with.>
>>
>The new dictionary will just be called "German" and will not have any
>description next to it.>
>>
>**I am after some feedback about whether I am doing the right thing
>with the new  "German" dictionary:**>
>>
>* An analysis of all unique characters in the current German dictionary
>shows these: `aÅäÄbcdefghijklmnoöÖŒpqrsŠßtuüÜvwxyz`>
>* Based on feedback from @d-mal in [this
>comment](https://github.com/lexica/lexica/issues/71#issuecomment-685184402),
>I am eplacing ä/Ä->ae, ö/Ö->oe, ü/Ä->ue, ß->ss. Note they are all
>ending up as lowercase because that is how Lexica stores dictionaries.
>Upon rendering them to screen, they will all be shown as upper case.>
>* Based on feedback from @DocOellerson in this issue, I'll replace Œ as
>oe. Indeed as mentioned above, this is only used in "Oeuvre" and
>"Oeuvres" in the current dictionary>
>* I will then remove any duplicate words in case this process
>accidentally introduced them.>
>* This leaves the following unique characters in the new dictionary:
>`aÅbcdefghijklmnopqrsŠtuvwxyz`.>
>>
>**My questions are:**>
>* **Does the above sound reasonable?**>
>* **What should be done with `Å` and `Š` (if anything?)**>
>* **Is it strictly incorrect to show them as upper case on the
>gameboard? Or is that just a matter of opinion?** Lexica does have the
>option of customizing the display of letters on a per-dictionary basis
>if we need to make these lower case but upper case other languages, but
>I'd rather leave it consistent if it is a correct option to do so.>
>* **Is there anything else we should do to improve this new slimline
>dictionary?**>
> >
>>
>For those technically minded, this is the script used to remove such
>characters is shown below (could probably be done better, but may as
>well just hack something together for our purposes) and it will be
>checked into this repository when done:>
>>
>```>
>echo "Copying source dictionary containing diacritics from
>$DICT_FILE_DIACRITICS...">
>cp "$DICT_FILE_DIACRITICS" "$DICT_FILE">
>cp "$LICENSE_FILE_DIACRITICS" "$LICENSE_FILE">
>>
>echo "Replacing ä/Ä->ae, ö/Ö->oe, ü/Ä->ue, ß->ss, Œ->oe...">
>sed -i "s/ä/ae/g" "$DICT_FILE">
>sed -i "s/Ä/ae/g" "$DICT_FILE">
>sed -i "s/ö/oe/g" "$DICT_FILE">
>sed -i "s/Ö/oe/g" "$DICT_FILE">
>sed -i "s/ü/ue/g" "$DICT_FILE">
>sed -i "s/Ü/ue/g" "$DICT_FILE">
>sed -i "s/ß/ss/g" "$DICT_FILE">
>sed -i "s/Œ/oe/g" "$DICT_FILE">
>```>
>>
>>
>-- >
>You are receiving this because you were mentioned.>
>Reply to this email directly or view it on GitHub:>
>https://github.com/lexica/lexica/issues/227#issuecomment-803654123

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by pserwylo** [comment]
Okay, here we go: [v2.1.0 has been tagged](https://github.com/lexica/lexica/releases/tag/v2.1.0) ready to be built by F-Droid. It includes both dictionaries as discussed in this thread. It doesn't include Weblate translations yet, but as they role in, I'll bump minor releases to include them (as they will have translations for these new dictionary descriptions) - just keen to get feedback on the actual dictionaries. Thanks again for all your input, this really wouldn't be such an international game withotu input from those in the community helping to ensure their language is supported.

**Comment by DocOellerson** [comment]
Excellent, thank you!! 

Am 27. März 2021 04:39:01 MEZ schrieb Peter Serwylo ***@***.***>:
>Closed #227.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/lexica/lexica/issues/227#event-4516623029

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.
